src/codeu/chat/server/Model.java

```diff
@@ -169,4 +228,178 @@ public void removeInterest(Uuid userId, Uuid interestId) {
       interests.get(userId).remove(interestId);
     }
   }
+  
+  public boolean getRestoredLog() {
+	return restoredLog;
+  }
+  
+  public void setRestoredLog(boolean restoredLog) {
+	this.restoredLog = restoredLog;
+  }
+  
+  public void refresh(File file) { 
+	// clear the current contents of the log
+	try {
+	  FileWriter fwOb = new FileWriter(file.getAbsolutePath(), false);
+	  PrintWriter pwOb = new PrintWriter(fwOb, false);
+	  pwOb.flush();
+	  pwOb.close();
+	  fwOb.close();
+	} catch (IOException e1) {
+      e1.printStackTrace();
```

Comment:
Fix indentation--don't mix tabs and spaces. If you haven't already, please configure your editor to emit spaces when you hit the tab key.